### PR TITLE
Update Playwright tests utilities to work both when the `new_header` flag is enabled and when it's disabled

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #1843 by @dhruvkb

## Description

This PR prepares the background for the tests with `new_header` enabled:
- Updates the `utils` methods to add the `mode` parameter. By default, it is set to `new`, and all the functions work with the `new_header`. The previous functions are used when `mode` parameter is set to `old`.
- Renames all the tests that use the affected `utils` methods to add `-old` suffix to them. The new tests without the suffix will be added in a follow-up PR.
- Adds the new breakpoint-related methods. Previously, desktop widths started at `md` breakpoint, and now they start at `lg`.
- Improves the accessible label for the `VContentSettingsButton` to add a number of filters applied.

## Testing Instructions

The tests in CI should pass.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: test/playwright/utils/navigation.ts [hunk 1]
```diff
@@ -171,27 +250,48 @@ export const assertCheckboxStatus = async (
...
+    await page.locator(`a[role="radio"]:has-text("${to}")`).click()
+    await closeMobileMenu(page, mode)
+  }
 }
 
 /**
- * Finds a button with a popup to the left of the filters button which doesn't have a 'menu' label
- * @param page - The current page
+ * For desktop, returns the content of the Content switcher button.
+ * For mobile, returns null.
```
Comment (obulat) [inline test/playwright/utils/navigation.ts:274]:
```suggestion
 * For mobile, returns the selected content type from the modal.
```

**Review by zackkrida — APPROVED** [review]
The `VContentSettingsButton` aria improvements look great as well. LGTM.

**Review by sarayourfriend — COMMENTED** [review]
I'm not always a huge fan of "old" and "new" as indicators for things like this. Whatever is "new" today eventually becomes old. I guess at that point we'd come back and refactor all of these to remove the "old" tests... but to someone reading this who isn't clued into the header work, these mode names are cryptic. Are there clearer names than new and old for the mode? Maybe even "new_header" or "old_header"?

Also, one nit, if we could extract the `'old' | 'new'` type out it would help with the eventual "delete the old mode" refactor... if you removed the shared type then TypeScript would tell you precisely everywhere that needed to be changed to accommodate the refactor :slightly_smiling_face:

**Comment by obulat** [comment]
> Maybe even "new_header" or "old_header"?

Also, one nit, if we could extract the 'old' | 'new' type out it would help with the eventual "delete the old mode" refactor... if you removed the shared type then TypeScript would tell you precisely everywhere that needed to be changed to accommodate the refactor 🙂

Great suggestions, thank you, @sarayourfriend! I've added the new type, and made `new_header` and `old_header` constants so that the value can be changed in a single place.

**Review by sarayourfriend — APPROVED** [review]
LGTM! The `-old` filenames would be nice to be `-with-old-header` instead, but just a nit pick and don't want to hold up the PR any more. Sorry I didn't think of it before.

Nice work getting this stuff consolidated :rocket:

**Pull request merged**